src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -344,52 +335,120 @@ static ucs_status_t uct_rdamcm_cm_ep_server_init(uct_rdmacm_cm_ep_t *cep,
                            uct_cm_ep_server_conn_notify_callback_t,
                            ucs_empty_function);
     if (status != UCS_OK) {
-        goto err;
+        goto err_reject;
     }
 
     cep->id->context = cep;
 
-    memset(&conn_param, 0, sizeof(conn_param));
-    conn_param.private_data = ucs_alloca(uct_rdmacm_cm_get_max_conn_priv() +
-                                         sizeof(uct_rdmacm_priv_data_hdr_t));
+    if (ucs_test_all_flags(params->field_mask,
+                           UCT_EP_PARAM_FIELD_PRIV_DATA |
+                           UCT_EP_PARAM_FIELD_PRIV_DATA_LENGTH)) {
+        priv_data        = params->private_data;
+        priv_data_length = params->private_data_length;
+    } else if (params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB) {
+        status = uct_rdmacm_cm_ep_pack_cb(cep, pack_priv_data,
+                                          &pack_priv_data_length);
+        if (status != UCS_OK) {
+            goto err_reject;
+        }
+
+        priv_data        = &pack_priv_data;
+        priv_data_length = pack_priv_data_length;
+    } else {
+        priv_data        = NULL;
+        priv_data_length = 0;
+    }
 
-    status = uct_rdmacm_cm_ep_pack_cb(cep, &conn_param);
+    status = uct_rdmacm_cm_ep_send_priv_data(cep, priv_data, priv_data_length);
     if (status != UCS_OK) {
-        goto err_reject;
+        goto err;
+    }
+
+    return uct_rdmacm_cm_ack_event(event);
+err_reject:
+    uct_rdmacm_cm_reject(cep->id);
```

Comment:
uct_rdmacm_cm_destroy_id ?

Reply:
it should be done in dtor which is caller of this function

Reply:
will the dtor be called if uct_rdamcm_cm_ep_server_init returns a non UCX_OK status?

Reply:
yes, `ctor` calls cleanup func if it's failed

Reply:
you are right, since cleans up starts from super, need to destroy id here